nixos/modules/services/misc/recyclarr.nix

```diff
@@ -0,0 +1,141 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.recyclarr;
+in
+{
+  options.services.recyclarr = {
+    enable = mkEnableOption "recyclarr service";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.recyclarr;
+      defaultText = literalExpression "pkgs.recyclarr";
+      description = "The recyclarr package to use.";
+    };
+
+    command = mkOption {
+      type = types.str;
+      default = "sync";
+      description = "The recyclarr command to run (e.g., sync).";
+    };
+
+    configFile = mkOption {
```

Comment:
Use RFC-42-style `settings` please. If you're worried about secrets, I'd recommend using `utils.genJqSecretsReplacementSnippet`.

Reply:
Would the expectation be that a YAML file is generated based on the given settings, and its path is passed as the `--config` argument to the binary?

Reply:
Yeah, you can look at the PR I made to `nginx-sso` as an example.

Reply:
With this approach, the file generation is moved into the service and out of the user's hands, making it cumbersome to use tools like SOPS or similar that are way better at managing secrets. Are we sure this is the correct solution in this case?

Reply:
That's the opposite: using `utils.genJqSecretsReplacementSnippet` makes it very easy to integrate with agenix/sops-nix to both set the configuration in nix while leaving secret handling to those tools.

I'm not sure I follow what is troubling you about it.

Reply:
The lack of documentation regarding the use of `genJqSecretsReplacementSnippet` made it unclear how it works. I will apply the change and conduct a test usage.

Reply:
Yeah, not sure why its documentation isn't surfaced in the manual/better documented. If the current usage in other modules isn't enough "documentation", I can walk you through its usage

Reply:
I applied the change and verified that the service is still working. I had to loosen some security permissions, but nothing major.